core/src/types/scheme.rs

```diff
@@ -134,6 +135,99 @@ impl Scheme {
     pub fn into_static(self) -> &'static str {
         self.into()
     }
+
+    /// Get all enabled schemes.
+    ///
+    /// OpenDAL could be compiled with different features, which will enable different schemes.
+    /// This function returns all enabled schemes so users can make decisions based on it.
+    pub fn enabled() -> HashSet<Scheme> {
```

Comment:
Does the naming here need to be more specific?

Reply:
Any suggestions?

Reply:
I didn't see how to call it. If you decide where and how it will be used, I think we can settle on a name.

Reply:
I think the name is clear enough:

```rust
let enabled_schemes = Scheme::enabled();
if !enabled_schemes.contains(Scheme::S3) {
   return Err(anyhow!("s3 support is not enabled"))
}
```

Reply:
If that's the case, I think it's good enough.

Reply:
I've added this code snippet to the examples.